PetaPoco/Core/PocoData.cs

```diff
@@ -125,7 +125,8 @@ public static PocoData ForType(Type type, IMapper defaultMapper)
             if (type == typeof(System.Dynamic.ExpandoObject))
                 throw new InvalidOperationException("Cannot use dynamic types with this method");
 
-            return _pocoDatas.GetOrAdd(type, () => new PocoData(type, defaultMapper));
+            var key = $"{defaultMapper.GetType().FullName}, {type.FullName}";
+            return _pocoDatas.GetOrAdd(key, () => new PocoData(type, defaultMapper));
```

Comment:
**issue:** Need to do `var mapper = Mappers.GetMapper(Type, defaultMapper);` here and use that mapper in the key.

**question:** Are there any pros/cons to using a string key like this vs. a tuple of type and mapper type?

Reply:
Pros: There is no need to keep the instances around (memory) and comparison is cheaper (cpu).
Cons: There could be collisions (although unlikely since we use fully qualified type names). We take a hit on readability and maintainability.

Let me use a tuple rather. There'll be consistency on the caches and the performance hit is minute.